### PR TITLE
Record scan-gap-interpolation in Landsat L1 metadata

### PR DESCRIPTION
**Pull request by jeremyh**
And bump pre-commit libraries, as the current version of flake8 is failing on some systems.

### CODE REVIEW

File: eodatasets3/properties.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ def parsed_sentinel_tile_id(tile_id) -> Tuple[str, Dict]:
     "landsat:image_quality_oli": None,
     "landsat:image_quality_tirs": None,
     "landsat:processing_software_version": None,
+    "landsat:scan_gap_interpolation": float,
```
Comment (sixy6e) [inline eodatasets3/properties.py:183]: Can you confirm if *float* is correct for the scan gap interpolation property?  I thought it'd be a boolean or something.

Comment (sixy6e) [reply eodatasets3/properties.py:183]: Is it number of pixels or something?

**Review by sixy6e — APPROVED** [review]
Looks good to me.

File: eodatasets3/properties.py (hunk 1 above)
Comment (jeremyh) [reply eodatasets3/properties.py:183]: Yeah, in the lpgs MTLs it's a float. 

In our collections, they're all either '2.0' (usgs) or '0.0' (ours). 

A boolean is probably more in line with how we use the field ourselves today, but it didn't seem worth deviating from the norm.

**Pull request merged**